### DISCUSSION TITLE
New GraphQL Query for LIST/array of images (metafield)

### DISCUSSION THREAD

**Discussion by davidmedero** [Shopify/storefront-api-feedback#144] [Ideas & Requests]
Please refer to this post:

https://community.shopify.com/c/storefront-api-and-sdks/what-s-the-grpahql-query-for-image-originalsrc-in-metafield-of/m-p/1754913#M2456

I can easily get the originalSrc like this:

            metafield(namespace: "custom", key: "color") {
              reference {
                ... on MediaImage {
                  image {
                    originalSrc
                  }
                }
              }
            }
 

But this doesn't work with a list (array) of images

At the moment, I'm forced to add an image metafield to each variant. It would be much more convenient for the user to simply add all variant images to the first variant/index all at once instead of having to go through each one and possibly even be compelled to add more 'fake' variants just to add more variant images...

**Comment by dominikzogg** [comment]
```
metafield(namespace: "custom", key: "color") {
          references(first: 10) {
            nodes {
              __typename
              ... on MediaImage {
                image {
                  originalSrc
                }
              }
            }
          }
        }
``

**Comment by davidmedero** [comment]
Thank you sooooooooo much @dominikzogg !!!!! You've helped me so much, I really appreciate it.

**Comment by dominikzogg** [comment]
@davidmedero i had the issue and found your issue before the solution on another page, so i remembered that i should post it here.
